core/src/test/java/nav/statusplattform/core/repositories/RecordRepositoryTest.java

```diff
@@ -290,4 +290,12 @@ private RecordEntity getRandomizedRecordEntity() {
                 .setResponsetime(SampleData.getRandomResponseTime());
     }
 
+    private List<RecordEntity> generateRandomizedRecordEntities(ServiceEntity serviceEntity, int amount) {
+        List<RecordEntity> records = new ArrayList<>();
+        for (int i = 0; i < amount; i++) {
```

Comment:
```suggestion
    private List<RecordEntity> generateRandomizedRecordEntities(ServiceEntity serviceEntity, int size) {
        if (size <= 0) {
            throw IllegalArgumentException("`size` argument refers to size of list, must be non-zero positive integer.");
        }
        List<RecordEntity> records = new ArrayList<>();
        for (int i = 0; i < size; i++) {
```

This edge case was not considered here. Not sure if suggested code compiles.